### PR TITLE
External semaphore compatibility fixes

### PR DESCRIPTION
**Pull request by mikes-lunarg**
Track when a semaphore payload is imported and restored, turn off
semaphore validation when using an external payload.

VK_KHR_external_semaphore_fd entry points and test cases to come in a follow-on patch

### CODE REVIEW

**Comment by chrisforbes** [comment]
Hmmn, this one is going to be fun. Give me some time to wrap my head around all the details here.

**Comment by chrisforbes** [comment]
@mikes-lunarg do you want to split out the two refactoring commits into their own pull request and land those, while we're still figuring out what to do with the actual external fence stuff?

**Comment by mikes-lunarg** [comment]
Rebased and added suggestion from @chrisforbes about retiring work early when an external semaphore is signaled

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -2692,8 +2693,18 @@ static void PostCallRecordQueueSubmit(layer_data *dev_data, VkQueue queue, uint3
                     pSemaphore->in_use.fetch_add(1);
                     semaphore_signals.push_back(semaphore);
                 } else {
-                    semaphore_externals.push_back(semaphore);
-                    pSemaphore->in_use.fetch_add(1);
+                    // Retire work up until this submit early, we will not see the wait that corresponds to this signal
+                    early_retire_seq = pQueue->seq + pQueue->submissions.size() + 1;
+                    static bool early_retire_warning = false;
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: Make this device state.

Comment (mikes-lunarg) [reply layers/core_validation.cpp]: Moved the warning bool into dev_data

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -2715,6 +2726,10 @@ static void PostCallRecordQueueSubmit(layer_data *dev_data, VkQueue queue, uint3
                                          submit_idx == submitCount - 1 ? fence : VK_NULL_HANDLE);
     }
 
+    if (early_retire_seq) {
+        RetireWorkOnQueue(dev_data, pQueue, early_retire_seq);
```
Comment (chrisforbes) [inline layers/core_validation.cpp]: Have you thought about whether we need to also call `VerifyQueueStateToSeq` before we do the retirement here? [As a bigger picture thing, I'm not sure the remaining error produced by  `VerifyQueueStateToSeq` is any good in general...]

Comment (chrisforbes) [reply layers/core_validation.cpp]: I think this is pretty much OK not calling that function, but something to think about.

**Pull request merged**